### DISCUSSION TITLE
Embellish selected column

### DISCUSSION THREAD

**Discussion by atagen** [YaLTeR/niri#448] [Ideas]
I'm a big fan of the "floating master" type layout that can be achieved in qtile - something like this.
![image](https://github.com/YaLTeR/niri/assets/11548989/7819268d-0206-479a-b4ee-3c2e0a5636fe)
It effectively scrolls like a carousel.

Would it be possible to emulate this kind of thing in Niri by "popping out" the selected column, increasing its height beyond the other columns, and widening it to overlap them on the sides?

This would be very visually impressive, and quite useful as a gross visual indicator of selection.

**Comment by YaLTeR** [comment]
Hm, idk. I'd like to avoid hardcoding this kind of stuff and I'm not sure if there's a theoretical window rule combination that could do this.

**Comment by atagen** [comment]
I understand.

I did note the discussion on a niri events ipc subscription command - perhaps this would work best as an external program that listens to focus change events and sends the appropriate resizes, similar to various autotiling solutions that exist for sway.

 I guess the overlapping stuff might have to wait for the floating window implementation too, but that's fine.

Btw, apologies, didn't realise responding via email posted all the reply chain junk..
